### ISSUE TITLE
Stream Management policy-violation "h attribute too big"

### ISSUE THREAD

**Issue by scatterbrain** [esl/MongooseIM#408]
I'm testing out Stream Management support and wondering why I'm receiving a policy-violation error from MongooseIM. I'm using MongooseIM 1.5.

Message Log from the client's point of view starting from the bind: 

```
XMPP Recv: <iq id="MX_1" type="result" xmlns="jabber:client">
  <bind xmlns="urn:ietf:params:xml:ns:xmpp-bind">
    <jid>a84d2e1007dd3d38379e3f7e5b200225c2f85f3ad72753febe4c4ce7aa00c6b4@testdomain.net/MatriX</jid>
  </bind>
</iq>


XMPP Send: <enable xmlns="urn:xmpp:sm:3" />

XMPP Send: <iq id="MX_2" type="set" xmlns="jabber:client">
  <session xmlns="urn:ietf:params:xml:ns:xmpp-session" />
</iq>

XMPP Recv: <enabled xmlns="urn:xmpp:sm:3" />

XMPP Recv: <iq type="result" id="MX_2" xmlns="jabber:client">
  <session xmlns="urn:ietf:params:xml:ns:xmpp-session" />
</iq>

XMPP Send: <iq id="MX_3" type="get" xmlns="jabber:client">
  <query xmlns="jabber:iq:roster" />
</iq>

XMPP Recv: <iq from="a84d2e1007dd3d38379e3f7e5b200225c2f85f3ad72753febe4c4ce7aa00c6b4@testdomain.net" to="a84d2e1007dd3d38379e3f7e5b200225c2f85f3ad72753febe4c4ce7aa00c6b4@testdomain.net/MatriX" id="MX_3" type="result" xmlns="jabber:client">
  <query xmlns="jabber:iq:roster”>
    <item subscription="none" jid="test@testdomain.net" />
  </query>
</iq>

XMPP Recv: <r xmlns="urn:xmpp:sm:3" />

XMPP Send: <a h="2" xmlns="urn:xmpp:sm:3" />

XMPP Recv: <stream:error xmlns:stream="http://etherx.jabber.org/streams">
  <policy-violation xmlns="urn:ietf:params:xml:ns:xmpp-streams" />
  <text xmlns="urn:ietf:params:xml:ns:xmpp-streams">h attribute too big</text>
</stream:error>
```

The xep-0198 states: 

>> There are two values of 'h' for any given stream: one maintained by the client to keep track of stanzas it has handled from the server, and one maintained by the server to keep track of stanzas it has handled from the client. The client initializes its value to zero when it sends `<enable/>` to the server, and the server initializes its value to zero when it sends `<enabled/>` to the client (it is expected that the server will respond immediately to `<enable/>` and set its counter to zero at that time). After this initialization, the client increments its value of 'h' for each stanza it handles from server, and the server increments its value of 'h' for each stanza it handles from the client."

In the above log, the client receives two `<iq/>` result stanzas after it has sent the `<enable/>` so if I understand this correctly, it has replied with the correct `h="2"` attribute to the server's `<r/>`

Why does the server complain about policy-violation?

**Comment by lavrin** [comment]
Good catch, thanks! We'll need to have a closer look at why this happens.

**Comment by studzien** [comment]
There was an issue with such an order of session establishment (``<enable>`` before ``<session>``).
The fix has already been merged into the master branch.
Please try if this solves your problem.

**Comment by michalwski** [comment]
The mentioned fix was done in #347

**Comment by scatterbrain** [comment]
You are correct, building from the master branch fixed this issue for me.
